sdks/java/core/src/main/java/org/apache/beam/sdk/io/FileSystem.java

```diff
@@ -102,8 +107,52 @@ protected abstract WritableByteChannel create(ResourceIdT resourceId, CreateOpti
    *     resource might or might not be copied. In such scenarios, callers can use {@code match()}
    *     to determine the state of the resources.
    */
-  protected abstract void copy(List<ResourceIdT> srcResourceIds, List<ResourceIdT> destResourceIds)
-      throws IOException;
+  @Deprecated
+  protected void copy(List<ResourceIdT> srcResourceIds, List<ResourceIdT> destResourceIds)
+      throws IOException {
+    throw new UnsupportedOperationException("deprecated - do not implement this method");
+  }
+
+  /**
+   * Copies a {@link List} of file-like resources from one location to another.
+   *
+   * <p>The number of source resources must equal the number of destination resources. Destination
+   * resources will be created recursively.
+   *
+   * @param srcResourceIds the references of the source resources
+   * @param destResourceIds the references of the destination resources
+   * @param moveOptions additional settings passed to the underlying implementation
+   * @throws FileNotFoundException if the source resources are missing. When copy throws, each
+   *     resource might or might not be copied. In such scenarios, callers can use {@code match()}
+   *     to determine the state of the resources.
+   */
+  protected void copy(
```

Comment:
I kinda see where you are going with having an Options object for file operations, But we should probably think more in detail what are this options for, e.g. having `MoveOptions` all around make sense ?

Reply:
Here I was copying CreateOptions, which is passed to `FileSystem.create()` implementations.
Were you thinking of having seperate FooOptions classes for each operation? Or a single class for all?